### PR TITLE
Fix(support npu): some little bugs for npu support

### PR DESCRIPTION
**Pull request by li126com**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: internlm/model/metrics.py [hunk 1]
```diff
@@ -153,6 +153,7 @@ def update(self, logits, labels, type_ids=None):
 
             acc = corrects.sum()
             torch.distributed.all_reduce(acc, op=torch.distributed.ReduceOp.SUM, group=self.tp_pg)
+            torch.npu.synchronize()
             # The synchronization here is to prevent unpredictable HANG when the NPU is running.
```
Comment (SolenoidWGT) [inline internlm/model/metrics.py:158]: 157行和156行交换下位置

**Comment by sunpengsdu** [comment]
image
also another one

File: internlm/model/metrics.py (hunk 1 above)
Comment (sunpengsdu) [reply internlm/model/metrics.py:158]: 使用stream的方式进行synchronize？因为GPU上我们直接这么用就会让overlap broadcast失效

Comment (sunpengsdu) [reply internlm/model/metrics.py:158]: > 使用stream的方式进行synchronize？因为GPU上我们直接这么用就会让overlap broadcast失效

@li126com see this

File: internlm/train/pipeline.py [hunk 2]
```diff
@@ -299,7 +299,7 @@ def initialize_optimizer(model: Union[nn.Module, nn.ModuleList], isp_communicato
     adam_extra_kwargs = {}
     # set fused=True to avoid nan grad norm when model size is larger and use_fp32_norm=True
 
-    if internlm_accelerator.get_accelerator_backend() == AcceleratorType.NPU:
+    if get_accelerator().get_accelerator_backend() == AcceleratorType.NPU:
```
Comment (sunpengsdu) [inline internlm/train/pipeline.py]: internlm_accelerator = get_accelerator() has already been defined, so no need to use get_accelerator()?

File: internlm/model/metrics.py [hunk 3]
```diff
@@ -155,7 +155,7 @@ def update(self, logits, labels, type_ids=None):
 
             acc = corrects.sum()
             torch.distributed.all_reduce(acc, op=torch.distributed.ReduceOp.SUM, group=self.tp_pg)
-            torch.npu.synchronize()
+            internlm_accelerator.synchronize()
```
Comment (sunpengsdu) [inline internlm/model/metrics.py]: internlm_accelerator = get_accelerator() has already been defined, so no need to use get_accelerator()?

File: internlm/model/metrics.py (hunk 1 above)
Comment (sunpengsdu) [reply internlm/model/metrics.py:158]: only sych when using NPU

**Review by sunpengsdu — APPROVED** [review]
LGTM

**Pull request merged**